packages/find-root/src/index.ts

```diff
@@ -98,50 +118,51 @@ function hasWorkspacesConfiguredViaPkgJsonSync(
   }
 }
 
-function hasWorkspacesConfiguredViaLernaSync(directory: string) {
-  try {
-    let lernaJson = fs.readJsonSync(path.join(directory, "lerna.json"));
-    if (lernaJson.useWorkspaces !== true) {
-      return directory;
-    }
-  } catch (err) {
-    if (!isNoEntryError(err)) {
-      throw err;
-    }
-  }
-}
+export function findRootSync(cwd: string): MonorepoRoot {
+  let monorepoRoot: MonorepoRoot | undefined;
+
+  findUpSync(
+    (directory) => {
+      const tools = defaultOrder.map((toolType) => supportedTools[toolType]);
 
-function hasWorkspacesConfiguredViaPnpmSync(directory: string) {
-  // @ts-ignore
-  let pnpmWorkspacesFileExists = fs.existsSync(
-    path.join(directory, "pnpm-workspace.yaml")
+      for (const tool of tools) {
+        if (tool.isMonorepoRootSync(directory)) {
+          monorepoRoot = {
+            tool: tool,
+            rootDir: directory,
+          };
+          return directory;
+        }
+      }
+    },
+    { cwd, type: "directory" }
   );
-  if (pnpmWorkspacesFileExists) {
-    return directory;
-  }
-}
 
-export function findRootSync(cwd: string) {
+  if (monorepoRoot) return monorepoRoot;
+
+  // If there is no monorepo root, but we can find a single package json file, we will
+  // return a "RootTool" repo, which is the special case where we just have a root package
+  // with no monorepo implementation (i.e.: a normal package folder).
```

Comment:
how does it play with the root tool being in the `defaultOrder` **and** with the fact that it's not even the last tool there? I didn't have time to analyze the situation to conclude why this is the case - but it definitely caught my eye and interest

Reply:
Fixed -- this was just me mis-ordering them, I think we want RootTool to always come last (technically not required, but makes the most sense to the reader I think).

Reply:
So the `RootTool` is still in the `defaultOrder` etc but it actually can't ever be matched through that cause it just returns `false` from `isMonorepoRoot`. Is it even worth keeping it there then?